### PR TITLE
Improve code quality

### PR DESCRIPTION
**Pull request by felipediel**
## Proposed changes
- Make python3 official.
- Add docstrings and annotations.
- Move calculate_crc16() to helpers.py.
- Create a separate file for each device type. Use the same file structure that we are using in Home Assistant, so it is easier to keep everything up to date.

## For reviewers
I did not change the logic of the code. I just added docstrings, function annotations and moved things. If you follow commit by commit it will be easier to understand what I am doing, so you don't need to review 1066 lines.

### CODE REVIEW

**Comment by felipediel** [comment]
@mjg59 @Danielhiversen What do you think of creating separate files for the base class (device.py) and its concrete implementations (switch.py, remote.py, climate.py, light.py, cover.py)? It would be much easier to keep control of things that should have the same interface and it would help a lot with annotations (we cannot annotate these classes in the current implementation).

**Comment by felipediel** [comment]
I'll clean up the legacy byte conversions in another PR, so we keep things focused.

File: broadlink/remote.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+# For legacy compatibility - don't use this
+class rm2(rm):
+    def __init__(self):
+        device.__init__(self, None, None, None)
+
+    def discover(self):
+        dev = discover()
+        self.host = dev.host
+        self.mac = dev.mac
```
Comment (felipediel) [inline broadlink/remote.py:132]: Do we still need this?

Comment (mjg59) [reply broadlink/remote.py:132]: There's some chance that someone has legacy code that still expects the original behaviour, but I wouldn't prioritise it. I'm fine with making breaking changes at this stage.

**Comment by mjg59** [comment]
> @mjg59 @Danielhiversen What do you think of creating separate files for the base class (device.py) and its concrete implementations (alarm.py, switch.py, remote.py, climate.py, light.py, sensor.py and cover.py)?
> 
> * It is easier to find things, solve problems and keep updates focused.
> * It is easier to keep control of things that should have the same interface.
> * It helps with annotations.
> * It helps to maintain and grow the Broadlink Integration in Home Assistant. I chose the same file structure we use there. So people can have a look at these files, see what is missing there and add support for new device types. We will have this library as a north. How about that?

Yes, I think this makes sense.

File: broadlink/remote.py (hunk 1 above)
Comment (felipediel) [reply broadlink/remote.py:132]: Great, thank you! I will leave breaking changes for an exclusive PR, so it is easier to revert if necessary.

**Pull request merged**